src/commonMain/kotlin/com.adamratzman.spotify/Builder.kt

```diff
@@ -334,29 +339,29 @@ class SpotifyAppApiBuilder(
     override var options: SpotifyApiOptions = SpotifyApiOptionsBuilder().build()
 ) : ISpotifyAppApiBuilder {
     /**
-     * Create a new [SpotifyAppAPI] that only has access to *public* endpoints and data
+     * Create a new [SpotifyAppApi] that only has access to *public* endpoints and data
      */
     fun buildPublic() = build()
 
     /**
-     * Build a public [SpotifyAppAPI] using the provided credentials
+     * Build a public [SpotifyAppApi] using the provided credentials
      *
      * Provide a consumer object to be executed after the api has been successfully built
      */
-    override fun buildAsync(consumer: (SpotifyAPI) -> Unit) = Thread { consumer(build()) }.start()
+    override fun buildAsync(consumer: (SpotifyApi) -> Unit) = GlobalScope.launch {
+        withContext(Dispatchers.Default) { consumer(build()) }
+    }
 
     /**
-     * Build a public [SpotifyAppAPI] using the provided credentials
+     * Build a public [SpotifyAppApi] using the provided credentials
      */
-    override fun build(): SpotifyAPI {
+    override fun build(): SpotifyApi {
         val clientId = credentials.clientId
         val clientSecret = credentials.clientSecret
-        if ((clientId == null || clientSecret == null) && (authorization.token == null && authorization.tokenString == null)) {
-            throw IllegalArgumentException("You didn't specify a client id or client secret in the credentials block!")
-        }
+        require(!((clientId == null || clientSecret == null) && (authorization.token == null && authorization.tokenString == null))) { "You didn't specify a client id or client secret in the credentials block!" }
```

Comment:
Well, it would be easier to understand if there wasn't a `!`. The term should be inverted.

Reply:
I won't make that on all occurrences 😉.